.github/workflows/bindings_python.yml

```diff
@@ -111,15 +111,14 @@ jobs:
     name: Release
     runs-on: ubuntu-latest
     if: "startsWith(github.ref, 'refs/tags/') && !contains(github.ref, '-')"
+    permissions:
+      contents: read
+      id-token: write
     needs: [macos, linux, windows]
     steps:
       - uses: actions/download-artifact@v3
         with:
           name: wheels
-      - name: Publish to PyPI
-        uses: PyO3/maturin-action@v1
-        env:
-          MATURIN_PYPI_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
+      - uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
I'm not sure if `pypa/gh-action-pypi-publish@release/v1` approved by the ASF. Would like to add a quick demo job that only have `pypa/gh-action-pypi-publish@release/v1`? I expect it to fail instead of skipped.

Reply:
![image](https://github.com/apache/incubator-opendal/assets/5351546/89453522-d43d-4ac8-92ea-6d1f5ea0bd22)

Reply:
Looks good to me, let's go! Thanks for your patience!